avalanche/logging/wandb_logger.py

```diff
@@ -4,48 +4,80 @@
 # See the accompanying LICENSE file for terms.                                 #
 #                                                                              #
 # Date: 25-11-2020                                                             #
-# Author(s): Vincenzo Lomonaco, Lorenzo Pellegrini                             #
+# Author(s): Diganta Misra, Andrea Cossu                                       #
 # E-mail: contact@continualai.org                                              #
 # Website: www.continualai.org                                                 #
 ################################################################################
-
 """ This module handles all the functionalities related to the logging of
 Avalanche experiments using Weights & Biases. """
-from PIL.Image import Image
+
+from typing import Union
+from pathlib import Path
+import os
+import errno
+
+import numpy as np
 from numpy import array
 from torch import Tensor
+import torch
+from torch import Tensor
+
+from PIL.Image import Image
 from matplotlib.pyplot import Figure
 
 from avalanche.evaluation.metric_results import AlternativeValues, \
     MetricValue, TensorImage
 from avalanche.logging import StrategyLogger
-import numpy as np
 
 
 class WandBLogger(StrategyLogger):
     """
     The `WandBLogger` provides an easy integration with
     Weights & Biases logging. Each monitored metric is automatically
     logged to a dedicated Weights & Biases project dashboard.
+
+    External storage for W&B Artifacts (for instance - AWS S3 and GCS
+    buckets) uri are supported.
+
+    The wandb log files are placed by default in "./wandb/" unless specified.
+
+    .. note::
+        TensorBoard can be synced on to the W&B dedicated dashboard.
     """
 
     def __init__(self, project_name: str = "Avalanche", 
-                 run_name: str = "Avalanche Test", params: dict = None):
+                 run_name: str = "Test", log_artifacts: bool = False,
+                 path: Union[str, Path] = "Checkpoints", 
+                 uri: str = None, sync_tfboard: bool = False, 
+                 save_code: bool = True, config: object = None, 
+                 dir: Union[str, Path] = None, params: dict = None):
         """
         Creates an instance of the `WandBLogger`.
-
-        :param project_name: Name of the W&B project.:
-        :param run_name: Name of the W&B run.:
+        :param project_name: Name of the W&B project.
+        :param run_name: Name of the W&B run.
+        :param log_artifacts: Option to log model weights as W&B Artifacts.
+        :param path: Path to locally save the model checkpoints.
+        :param uri: Reference to external URI.
```

Comment:
could you expand a bit on this parameter?

Reply:
The `uri` parameter?

Reply:
exactly

Reply:
Sure, I'll add a more detailed definition of that parameter.